cli/pkg/cmdutil/githelper.go

```diff
@@ -0,0 +1,200 @@
+package cmdutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/semaphore"
+)
+
+var (
+	gitignoreHasDotenvRegexp       = regexp.MustCompile(`(?m)^\.env$`)
+	gitignoreHasDotRillCloudRegexp = regexp.MustCompile(`(?m)^\s*\.rillcloud/`)
+)
+
+// GitHelper manages git operations for a project
+// It also caches the git credentials for the project
+type GitHelper struct {
+	c         *client.Client
+	org       string
+	project   string
+	localPath string
+
+	// do not access gitConfig directly, use FetchGitConfig and setGitConfig
+	gitConfig   *gitutil.Config
+	gitConfigMu *semaphore.Weighted
+}
+
+func NewGitHelper(adminClient *client.Client, org, project, localPath string) *GitHelper {
```

Comment:
Consider refactoring this to `(h *Helper) GitHelper(project, localPath string)` (org is already configured on the `Helper` and the client should be retrieved lazily from the helper).